### ISSUE TITLE
`tuneup check` reports problems, but `dartanalyzer` doesn't

### ISSUE THREAD

**Issue by zoechi** [google/tuneup.dart#34]
```
Checking project bwu_polymer_routing...
3 issues found; analyzed 9 source files in 19.2s.

[warning] The argument type 'BindView (/home/myuser/dart/bwu_polymer_routing/lib/bind_view.dart)' cannot be assigned to the parameter type 'BindView (/home/myuser/dart/bwu_polymer_routing/lib/bind_view.dart)' at lib/bind_view.dart, line 46.
[warning] The argument type 'BindView (/home/myuser/dart/bwu_polymer_routing/lib/bind_view.dart)' cannot be assigned to the parameter type 'BindView (/home/myuser/dart/bwu_polymer_routing/lib/bind_view.dart)' at lib/bind_view.dart, line 61.
[info] The argument type 'View (/home/myuser/dart/bwu_polymer_routing/lib/src/routing.dart)' cannot be assigned to the parameter type 'View (/home/myuser/dart/bwu_polymer_routing/lib/src/routing.dart)' at lib/src/routing.dart, line 163.
```

source: https://github.com/bwu-dart/bwu_polymer_routing
Dart VM version: 1.11.0-edge.c67f93099fa3ac3e6a80e08517c16fe0df842806 (Mon Jun  8 18:22:56 2015) on "linux_x64"
`grind analyze` runs the analyzer

**Comment by devoncarew** [comment]
Thanks for the repro case!

I think you are importing the same library through two different paths - probably a package: import in one file, and a relative path import in another. I think the `dartanalyzer` command does some import normalization to handle cases like this. Which may or may not be the right thing; I think at runtime they _are_ actually two different libraries.

**Comment by zoechi** [comment]
This was what I assumed but wasn't able to find any suspicious import. I also moved the examples into a sub-project to get them out of the equation.

**Comment by astashov** [comment]
Another condensed example:

```dart
// lib/a.dart
library a;
import 'package:analyzer_bug/b.dart';

class Foo {}

main() {
  Foo foo = new Foo();
  bar(foo);
}

// lib/b.dart
library b;

import 'package:analyzer_bug/a.dart';

bar(Foo foo) {}

// pubspec.yaml
name: analyzer_bug
version: 0.0.1
description: A simple console application.
dependencies:
  tuneup: any
```

will show an error:

```
[warning] The argument type 'Foo (/Users/anton/projects/analyzer-bug/lib/a.dart)' cannot be assigned to the parameter type 'Foo (/Users/anton/projects/analyzer-bug/lib/a.dart)' at lib/a.dart, line 9.
```

Seems like a totally valid code, both imports are the same. Maybe the problem with circular references?

**Comment by danschultz** [comment]
+1 for a fix

**Comment by devoncarew** [comment]
Looking into this now -

**Comment by devoncarew** [comment]
published `0.1.2`

**Comment by danschultz** [comment]
Thanks!
